### PR TITLE
Accepting subscription id as deployment parameter

### PR DESCRIPTION
**Pull request by nehalilani**
### Additions
1. Currently, only subscription name could be used to select a subscription.
    1. Made changes to accept both subscription name and subscription ID as input from user.
2. Trimming extra empty spaces from user entered text i.e. subscription Name/ID and Location Name.

### Testing
initial testing done

### CODE REVIEW

File: deployment/Deploy.ps1 [hunk 1]
```diff
@@ -89,15 +89,18 @@ process {
 
         function Set-LtiActiveSubscription {
             param (
-                [string]$Name,
+                [string]$NameOrId,
                 $List
             )
             
-            $subscription = ($List | Where-Object { $_.name -ieq $Name })
```
Comment (adbindal) [inline deployment/Deploy.ps1:96]: Could the following be considered?
``` powershell
$subscription = ($List | Where-Object { ($_.name -ieq $NameOrId) -or ($_.id -ieq $NameOrId) })
```

File: deployment/Deploy.ps1 [hunk 2]
```diff
@@ -112,21 +115,23 @@ process {
...
-        elseif ($SubscriptionName) {
-            Write-Log -Message "Using User provided Subscription Name: $SubscriptionName"            
+        elseif ($SubscriptionNameOrId) {
+            Write-Log -Message "Using User provided Subscription Name/ID: $SubscriptionNameOrId"            
         }
         elseif ($SubscriptionCount -eq 1) {
-            $SubscriptionName = $SubscriptionList[0].name;
-            Write-Log -Message "Defaulting to Subscription Name: $SubscriptionName"
+            $SubscriptionNameOrId = $SubscriptionList[0].id;
+            Write-Log -Message "Defaulting to Subscription Name/ID: $SubscriptionNameOrId"
```
Comment (adbindal) [inline deployment/Deploy.ps1:123]:
```suggestion
            Write-Log -Message "Defaulting to Subscription ID: $SubscriptionNameOrId"
```

File: deployment/Deploy.ps1 [hunk 3]
```diff
@@ -112,21 +115,23 @@ process {
...
         }
         else {
             $SubscriptionListOutput = $SubscriptionList | Select-Object @{ l="Subscription Name"; e={ $_.name } }, "id", "isDefault"
             Write-Host ($SubscriptionListOutput | Out-String)
-            $SubscriptionName = Read-Host 'Enter Subscription Name from Above List'
-            Write-Log -Message "User Entered Subscription Name: $SubscriptionName"
+            $SubscriptionNameOrId = Read-Host 'Enter Subscription Name/ID from Above List'
```
Comment (adbindal) [inline deployment/Deploy.ps1:128]: How about the following?
```suggestion powershell
            $SubscriptionNameOrId = Read-Host 'Enter the Name or ID of the Subscription from Above List'
```

**Comment by adbindal** [comment]
This should take care of #86, where we had duplicate subscription names and user was blocked without the ability to pass subscription id as user input.

File: deployment/Deploy.ps1 (hunk 1 above)
Comment (nehalilani) [reply deployment/Deploy.ps1:96]: incorporating in my next commit

**Review by adbindal — APPROVED** [review]
LGTM 👍

**Pull request merged**